src/Bicep.Core/Semantics/Namespaces/NamespaceResolver.cs

```diff
@@ -61,12 +60,17 @@ void TryAddBuiltInNamespace(string @namespace)
             }
 
             TryAddBuiltInNamespace(SystemNamespaceType.BuiltInName);
-            if (sourceFile.FileKind == BicepSourceFileKind.BicepFile)
+
+            if (sourceFile.FileKind == BicepSourceFileKind.BicepFile) // don't register "az" namespace for Bicep Parameters files
             {
-                // don't register "az" namespace for Bicep Parameters files
-                TryAddBuiltInNamespace(AzNamespaceType.BuiltInName);
+                var foundAzProviderDeclaration = sourceFile.ProgramSyntax.Children
+                    .OfType<ImportDeclarationSyntax>()
+                    .Any(x => x.Alias is not null && x.Specification.Name.Equals(AzNamespaceType.BuiltInName));
```

Comment:
Need to remove the null check on `x.Alias` as it's optional.
```suggestion
                    .Any(x => x.Specification.Name.Equals(AzNamespaceType.BuiltInName, LanguageConstants.IdentifierComparison));
```